test/metabase/query_processor/middleware/annotate_test.clj

```diff
@@ -460,15 +460,14 @@
   (qp.store/with-metadata-provider meta/metadata-provider
     (testing (str "if a driver is kind enough to supply us with some information about the `:cols` that come back, we "
                   "should include that information in the results. Their information should be preferred over ours")
-      (is (=? {:cols [{:name           "totalEvents"
-                       :display_name   "Total Events"
+      (is (=? {:cols [{:display_name   "Total Events"
                        :base_type      :type/Text
                        :effective_type :type/Text
                        :source         :aggregation
                        :field_ref      [:aggregation 0]}]}
               (add-column-info
                (lib.tu.macros/mbql-query venues {:aggregation [[:metric 1]]})
-               {:cols [{:name "totalEvents", :display_name "Total Events", :base_type :type/Text}]}))))))
```

Comment:
Prior this PR metric name was slapped onto ... in .... In this case that is nil because metric 1 is not defined. 

Then, `metabase.query-processor.middleware.annotate/merge-col-metadata` picks `:name` from mocked cols over `nil`.

With the change to `lib.metadata.calculation/metadata-method :metric` in place the underlying logic sets the `:name` to `"nil"` for not defined metric.

The `metabase.query-processor.middleware.annotate/merge-col-metadata` then uses `"nil"`.

That said, it does not make sense to test this for undefined metric.